src/registration.jl

```diff
@@ -5,6 +5,9 @@ const registry = Dict{String, AbstractDataDep}()
 function register(datadep::AbstractDataDep)
     name = datadep.name
     if haskey(registry, name)
+        if isequal(datadep, registry[name])
+            return registry
```

Comment:
```suggestion
            return registry[name]
```

Might be a cleaner way to do this, using say:
 
```
existing = get(regisitry, name, nothing)
if existing != nothing
     if isequal(datadep, existing)
          return existing
    else
          @warning...
     end
end
```

or something like that?